password/impl/src/main/java/org/wildfly/security/password/impl/OneTimePasswordImpl.java

```diff
@@ -106,6 +109,28 @@ <S extends KeySpec> S getKeySpec(Class<S> keySpecType) throws InvalidKeySpecExce
         throw new InvalidKeySpecException();
     }
 
+    @Override
+    Password translate(final AlgorithmParameterSpec parameterSpec) throws InvalidKeyException, InvalidAlgorithmParameterException {
+        if (parameterSpec instanceof OneTimePasswordAlgorithmSpec) {
+            OneTimePasswordAlgorithmSpec updateSpec = (OneTimePasswordAlgorithmSpec) parameterSpec;
+            int updateSequence = updateSpec.getSequenceNumber();
+            if (updateSequence < this.sequenceNumber) {
+                throw new InvalidAlgorithmParameterException();
```

Comment:
I think it would be good to use `ElytronMessages` here and in the other places where an exception is thrown below.